integrations/c/workload-executor

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+
+echo "Running C Driver workload-executor..."
+
+declare script_dir
+script_dir="$(pwd)" || exit
+declare -r script_dir
+
+declare -r connection_string="${1:?}"
+declare -r workload_spec="${2:?}"
+
+# Populated via install-driver.sh.
+pushd mongo-c-driver || exit
+
+declare test_binary
+test_binary="$(pwd)/src/libmongoc/test-atlas-executor" || exit
+
+# Ensure required binary is present.
+command -V "${test_binary}" >/dev/null || exit
+
+export ASAN_OPTIONS="abort_on_error=1:symbolize=1"
+export UBSAN_OPTIONS="abort_on_error=1:print_stacktrace=1"
+export LSAN_OPTIONS="suppressions=.lsan-suppressions"
+
+export MONGOC_TEST_ATLAS="ON"
+export MONGOC_TEST_URI="${connection_string}"
+
+# Normally test-atlas-executor should always succeed and report failures/errors via generated files.
+# However, if it failed to do so (e.g. due to an assertion failure), communicate execution failure to astrolabe directly.
+if ! "${test_binary}" "${workload_spec}"; then
+  cat >|events.json <<DOC
```

Comment:
Is there a reason we need to manually build these JSON output files instead of exiting with a non-zero exit code and letting Astrolabe fail the task?

Reply:
To conform to the [workload executor specification]():

> The values of `numErrors` and `numFailures` are used by `astrolabe` to determine the overall success or failure of a driver workload execution. A non-zero value for either of these fields is construed as a sign that something went wrong while executing the workload and the test is marked as a failure. The workload executor’s exit code is **not** used for determining success/failure and is ignored.

I had interpreted this to mean that "exiting with a non-zero exit code" would not constitute a proper means of communicating failure (as documented). If astrolabe _does_ handle non-zero exit code for the workload executor script properly, then I believe we should be able to remove this.